mule/task/s3/__init__.py

```
@@ -47,39 +27,28 @@ def execute(self, job_context):
 class DownloadFile(ITask):
     required_fields = [
         'bucketName',
-        'objectName'
+        #        'prefix'
     ]
 
-    def __init__(self, args):
-        super().__init__(args)
-        self.bucketName = args['bucketName']
-        self.objectName = args['objectName']
-        self.outputDir = args['outputDir'] if 'outputDir' in args else '.'
-        self.fileName = args['fileName'] if 'fileName' in args else None
-
-    def download_file(self):
-        s3_util.download_file(self.bucketName, self.objectName, self.outputDir, self.fileName)
-
-    def execute(self, job_context):
-        super().execute(job_context)
-        self.download_file()
-
-
-class DownloadFiles(ITask):
-    required_fields = [
-        'bucketName',
-        'prefix'
-    ]
+#    optional_fields = [
+#        'fileName',
+#        'objectName'
+#    ]
 
     def __init__(self, args):
         super().__init__(args)
         self.bucketName = args['bucketName']
         self.prefix = args['prefix']
         self.suffix = args['suffix'] if 'suffix' in args else None
+        self.fileName = args['fileName'] if 'suffix' in args else None
+        self.objectName = args['objectName'] if 'suffix' in args else None
         self.outputDir = args['outputDir'] if 'outputDir' in args else '.'
 
     def download_files(self):
-        s3_util.download_files(self.bucketName, self.prefix, self.suffix, self.outputDir)
+        if self.objectName or self.filename:
```

Is this a potential error condition if the output file name is provided, but the object name is not?

---

I just tested and no error, it looks ok.

---

I just tested and no error, it looks ok.

---

Should you raise an error in this case? since the assumption is that there is one and only one object to download into the specified file name.

---

There are a couple of ways to download files:
1. Specify a file either as an object name or a filename.
1. If there is no #1, a prefix and/or suffix is expected.

I think you're right, there should be an exception raised if none of those conditions are met.  I'll add one.

---

Did you want to add an exception in case the object name was not specified?

---

I don't think so, because a `fileName` is the only field needed to download a file (along with `prefix`, which is required).

If both were specified, which would win?

---

Ah, nevermind that last comment.  I misunderstood the filename to be the name of the file that was wanted to be downloaded.

I'll fix that.